### ISSUE TITLE
Copying a justified text line from Word generated PDF results in new lines between characters

### ISSUE THREAD

**Issue by hfhchan** [mozilla/pdf.js#7791]
I'm seeing this issue with nearly every single export file from Word 2013.
I'm using Firefox 49.0.2 with built-in pdf.js.

Sample:
[test.pdf](https://github.com/mozilla/pdf.js/files/577724/test.pdf)

Copied text:
```
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
T
est
words
-
with
-
hyphenation
Test  words 
with
dashes 
–
test 
words with dashes
Test  words 
with
dashes 
–
test 
words  with dashes
Test  words 
with
dashes 
–
test 
words with dashes
Test words 
with
dashes 
–
test 
words with dashes
Test 
words 
with
dashes 
–
test 
words with dashes
Test words 
with
dashes 
–
test 
words with 
dashes
Test words 
with
dashes 
–
test 
words with dashes
```

Expected text:
```
Test words-with-hyphenation Test words-with-hyphenation Test words-with-
hyphenation Test words-with-hyphenation Test words-with-hyphenation Test words-
with-hyphenation Test words-with-hyphenation Test words-with-hyphenation Test 
words-with-hyphenation Test words-with-hyphenation Test words-with-hyphenation 
Test words-with-hyphenation Test words-with-hyphenation Test words-with-
hyphenation Test words-with-hyphenation Test words-with-hyphenation Test words-
with-hyphenation Test words-with-hyphenation Test words-with-hyphenation Test 
words-with-hyphenation Test words-with-hyphenation Test words-with-hyphenation 
Test words-with-hyphenation Test words-with-hyphenation Test words-with-
hyphenation Test words-with-hyphenation Test words-with-hyphenation Test words-
with-hyphenation Test words-with-hyphenation Test words-with-hyphenation Test 
words-with-hyphenation Test words-with-hyphenation Test words-with-hyphenation 
Test words-with-hyphenation Test words-with-hyphenation Test words-with-
hyphenation Test words-with-hyphenation Test words-with-hyphenation Test words-
with-hyphenation Test words-with-hyphenation Test words-with-hyphenation Test 
words-with-hyphenation Test words-with-hyphenation Test words-with-hyphenation 
Test words-with-hyphenation Test words-with-hyphenation Test words-with-
hyphenation Test words-with-hyphenation Test words-with-hyphenation Test words-
with-hyphenation Test words-with-hyphenation Test words-with-hyphenation Test 
words-with-hyphenation Test words-with-hyphenation Test words with dashes – test 
words with dashes Test words with dashes – test words with dashes Test words with 
dashes – test words with dashes Test words with dashes – test words with dashes Test 
words with dashes – test words with dashes Test words with dashes – test words with 
dashes Test words with dashes – test words with dashes
```

**Comment by timvandermeij** [comment]
This improved slightly after a recent fix (but it's still not really great). It now copies as:

```
Test
words
-with
-hyphenation
Test
words
-with
-hyphenation
Test
words
-with
-
hyphenation
  Test
  words
-with
-hyphenation
  Test
  words
-with
-hyphenation
  Test
  words
-
with
-hyphenation
  Test
  words
-with
-hyphenation
  Test
  words
-with
-hyphenation
  Test
words
-with
-hyphenation
  Test
  words
-with
-hyphenation
  Test
  words
-with
-hyphenation
Test
words
-with
-hyphenation
Test
words
-with
-hyphenation
Test
words
-with
-
hyphenation
  Test
  words
-with
-hyphenation
  Test
  words
-with
-hyphenation
  Test
  words
-
with
-hyphenation
  Test
  words
-with
-hyphenation
  Test
  words
-with
-hyphenation
  Test
words
-with
-hyphenation
  Test
  words
-with
-hyphenation
  Test
  words
-with
-hyphenation
Test
words
-with
-hyphenation
Test
words
-with
-hyphenation
Test
words
-with
-
hyphenation
  Test
  words
-with
-hyphenation
  Test
  words
-with
-hyphenation
  Test
  words
-
with
-hyphenation
  Test
  words
-with
-hyphenation
  Test
  words
-with
-hyphenation
  Test
words
-with
-hyphenation
  Test
  words
-with
-hyphenation
  Test
  words
-with
-hyphenation
Test
words
-with
-hyphenation
Test
words
-with
-hyphenation
Test
words
-with
-
hyphenation
  Test
  words
-with
-hyphenation
  Test
  words
-with
-hyphenation
  Test
  words
-
with
-hyphenation
  Test
  words
-with
-hyphenation
  Test
  words
-with
-hyphenation
  Test
words
-with
-hyphenation
  Test
  words
-with
-hyphenation
  Test
  words
-with
-hyphenation
Test
words
-with
-hyphenation
Test
words
-with
-hyphenation
Test
words
-with
-
hyphenation
  Test
  words
-with
-hyphenation
  Test
  words
-with
-hyphenation
  Test
  words
-
with
-hyphenation
  Test
  words
-with
-hyphenation
  Test
  words
-with
-hyphenation
  Test
words
-with
-hyphenation
  Test
  words
-with
-hyphenation
  Test  words 
with
  dashes 
–
  test 
words with dashes
 Test  words 
with
 dashes 
–
 test 
words  with dashes
 Test  words 
with
dashes 
–
 test 
words with dashes
 Test words 
with
 dashes 
–
 test 
words with dashes
 Test 
words 
with
 dashes 
–
 test 
words with dashes
 Test words 
with
 dashes 
–
 test 
words with 
dashes
 Test words 
with
 dashes 
–
 test 
words with dashes
```

**Comment by Snuffleupagus** [comment]
WFM, most likely fixed by PR #13257.
